dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hive/HiveMetadataConnector.java

```diff
@@ -380,6 +382,85 @@ protected String toCallDescription() {
     }
   }
 
+  @ParametersAreNonnullByDefault
+  private static class PartitionsJsonlTask extends AbstractHiveMetadataTask {
+
+    private PartitionsJsonlTask(Predicate<String> databasePredicate) {
+      super("partitions.jsonl", databasePredicate);
+    }
+
+    @Override
+    protected void run(Writer writer, ThriftClientHandle thriftClientHandle) throws Exception {
+      try (ThriftClientPool clientPool =
+          thriftClientHandle.newMultiThreadedThriftClientPool("partitions-task-pooled-client")) {
+        clientPool.execute(
+            thriftClient -> {
+              ImmutableList<String> allDatabases = thriftClient.getAllDatabaseNames();
+              for (String databaseName : allDatabases) {
+                if (isIncludedDatabase(databaseName)) {
+                  ImmutableList<String> allTables =
+                      thriftClient.getAllTableNamesInDatabase(databaseName);
+                  try (ConcurrentProgressMonitor monitor =
+                      new ConcurrentRecordProgressMonitor(
+                          "Writing partitions of tables in database '"
+                              + databaseName
+                              + "' to "
+                              + getTargetPath(),
+                          allTables.size())) {
+                    for (String tableName : allTables) {
+                      dumpPartitions(monitor, writer, clientPool, databaseName, tableName);
+                    }
+                  }
+                }
+              }
+            });
+      }
+    }
+
+    private void dumpPartitions(
+        ConcurrentProgressMonitor monitor,
+        Writer writer,
+        ThriftClientPool clientPool,
+        String databaseName,
+        String tableName) {
+      clientPool.execute(
+          (thriftClient) -> {
+            try {
+              monitor.count();
+              ImmutableList<? extends TBase<?, ?>> partitions =
+                  thriftClient.getTable(databaseName, tableName).getRawPartitions();
+              ThriftJsonSerializer jsonSerializer = new ThriftJsonSerializer();
+              synchronized (writer) {
+                JsonWriter jsonWriter = new JsonWriter(writer);
+                jsonWriter.setLenient(true);
+                writer.write("{\"databaseName\":");
+                jsonWriter.value(databaseName);
+                writer.write(",\"tableName\":");
```

Comment:
It looks like we have a global "all partitions" file here. Could we change that format to have a per table file in a single ZIP with partitions that from particular table? That would clearly help with scalability of reading this later.

Reply:
Done in https://github.com/google/dwh-migration-tools/pull/464